### ISSUE TITLE
[1.11] on Forge 12.20.0.2222 Weird Biome color when entering Biome Wasteland

### ISSUE THREAD

**Issue by Baalith** [Glitchfiend/BiomesOPlenty#949]
#### Issue Description:
Hi there,

I'm playing a little while with Biomes O Plenty on MC 1.11.2.
Everything works fine so far, exept the weird colors of the blocks when entering the biome Wasteland.
All other Biomes worked fine till today (ex. entering Swampland - the green Fog/Sky).
I've taken two screenshots to show the issue, hope it helps a bit. (see below)

#### Steps to reproduce:
Just walked into a wasteland biome.

#### Additional Information:

- Screenshots:

  - Standing in the Desert: Normal Color
![2017-02-12_11 14 34](https://cloud.githubusercontent.com/assets/4653580/22861354/90635bba-f117-11e6-8a61-cf61b67a2ccd.png)

  - Walked into the Wasteland: Weird Colors :/
![2017-02-12_11 14 28](https://cloud.githubusercontent.com/assets/4653580/22861355/99462474-f117-11e6-94a8-402fb272f76c.png)
____
#### Affected Versions (Exact numbers, do *not* use "latest"):

- Biomes O' Plenty: BiomesOPlenty-1.11.2-6.0.1.2145-universal
- Minecraft: 1.11.2
- Forge: 13.20.0.2222

**Comment by Adubbz** [comment]
Cannot reproduce this, have you tried with just BoP on its own? Things like this are usually caused by mods like Optifine

**Comment by Baalith** [comment]
Yeah, I tried it with BoP on its own. Strangely then the issue doesn't occur.
I noticed while browsing the source of BoP in the file:
[src/main/java/biomesoplenty/common/biome/overworld/BiomeGenWasteland.java](url)

at Line 57: this.addWeight(BOPClimates.WASTELAND, 50)

The number 50 seems a bit high, by comparison with the other biomes. (There the value differs between 1 and 10)
May this be the reason for the strange looking color with optifine installed? (Just a guess^^)

Best regards

**Comment by Syndaryl** [comment]
No, that's controlling how frequently you find a Wasteland.

**Comment by Baalith** [comment]
Okay, it just was a shot into the dark :)
Then it's surely a optifine issue (the strange color of the biome).
Strange for me is, all the other fog/sky colors from the other biomes seem to work with optifine, so what makes this thing so special?

**Comment by Forstride** [comment]
Yeah, I can't seem to reproduce this, even with OptiFine.

**Comment by Aphyxia** [comment]
This is caused by Dynamic Surroundings mod for me.
Without: [https://gyazo.com/4d9ac1ab58241250841dd2a8872b1adc](url)
With: [https://gyazo.com/0807b4d0bcd491a5f5d1f0fd38427663](url)
Dynamic surroundings biome fog options don't seem to affect this in any way.

**Comment by OreCruncher** [comment]
Not sure what in this combination is making things strange, but from a Dynamic Surroundings standpoint my RenderFogEvent handler is LOWEST priority and checks to see if the result is anything but default.  If default it handles, otherwise it defers to whatever high priority fog rendered decided.  I looked at the BoP code and do not see where it would set the result on the event if it handled.

(If I were to guess based on the screenshots the distance calcs are off.  During my trial and tribulation with my fog handler I noticed such strangeness when a value when negative or the fog "end" distance was closer to the player than "start".)
